monkestation/code/modules/mentor/mentorfriend.dm

```diff
@@ -0,0 +1,159 @@
+//Section for the Mentor Friend verb
+
+/mob/verb/imaginary_friend()
+	set category = "Mentor"
+	set name = "Become Imaginary Friend"
+
+	if(!usr.client.is_mentor())
+		return
+
+	if(istype(usr, /mob/camera/imaginary_friend/mentor))
+		var/mob/camera/imaginary_friend/mentor/mentorfriend = usr
+		mentorfriend.unmentor()
+		return
+
+	if(usr.client.is_mentor() && !isobserver(usr))
+		to_chat(usr, span_warning("You can only be an imaginary friend when you are observing."))
+		return
+
+	if(!isobserver(usr))
+		usr.client.admin_ghost()
+
+	var/mob/living/mentee
+	switch(input("Select by:", "Imaginary Friend") as null|anything in list("Key", "Mob"))
```

Comment:
If you are going to be checking for mobs with clients only, get the non-ghosts/non-new players in the player_list or alive_mob_list global.

Reply:
updated this, doesn't show as outdated, however.